### PR TITLE
Feat- Add to wishlist

### PR DESCRIPTION
**Pull request by abhij1607**
image

- added context for adding to wishlist and cart
- add to wishlist functionality
- counter badge in header/nav

### CODE REVIEW

File: src/context/user-products-context.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { createContext, useContext, useReducer } from "react";
+
+const UserProductsContext = createContext();
+
+const useUserProducts = () => useContext(UserProductsContext);
+
+const initialUserProducts = { wishlist: [], cart: [] };
+
+const userProductReducer = (state, action) => {
+  console.log(state, action);
```
Comment (Hamdrive) [inline src/context/user-products-context.jsx]: Cleanup console logs when raising PRs

File: src/pages/Wishlist/wishlist.js [hunk 2]
```diff
@@ -1,143 +1,54 @@
 import "./wishlist.css";
+import { useUserProducts } from "../../context/user-products-context";
 
 const Wishlist = () => {
+  const { productState, productDispatch } = useUserProducts();
+  console.log(productState);
```
Comment (Hamdrive) [inline src/pages/Wishlist/wishlist.js]: Cleanup console logs

File: src/pages/Wishlist/wishlist.js [hunk 3]
```diff
@@ -1,143 +1,54 @@
...
   return (
     <div className="wrapper">
       <main className="main-pane auto-container">
         <h1 className="h2">
           My Wishlist
-          <span className="p-lg txt-wt-light">6 Items</span>
+          <span className="p-lg txt-wt-light">
+            ` {productState.wishlist.length} Items`
```
Comment (Hamdrive) [inline src/pages/Wishlist/wishlist.js:12]: Not required to pass it within backticks, as JSX already considers it a string within HTML elements

**Review by Hamdrive — COMMENTED** [review]
Overall good work, especially implementing the context for wishlist and cart.

You have mentioned added context for cart functionality as well, but it does not seem to work in your deployment. Do look into that.

Additionally have suggested some cleanup and fixes based on deployment preview

File: src/context/user-products-context.jsx (hunk 1 above)
Comment (abhij1607) [reply src/context/user-products-context.jsx]: sure, thanks for pointing it

File: src/pages/Wishlist/wishlist.js (hunk 2 above)
Comment (abhij1607) [reply src/pages/Wishlist/wishlist.js]: sure, thanks for pointing it

File: src/pages/Wishlist/wishlist.js (hunk 3 above)
Comment (abhij1607) [reply src/pages/Wishlist/wishlist.js:12]: I needed to add the extra space to start, so used it...I'll refactor these styling related things, later on. Thanks

**Comment by abhij1607** [comment]
> 

I'll add the functionality of Add to Cart later in the next PR

File: src/context/user-products-context.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { createContext, useContext, useReducer } from "react";
+
+const UserProductsContext = createContext();
+
+const useUserProducts = () => useContext(UserProductsContext);
+
+const initialUserProducts = { wishlist: [], cart: [] };
+
+const userProductReducer = (state, action) => {
```
Comment (junaidshaikh-js) [inline src/context/user-products-context.jsx:9]: type and payload can be destructured here 

`( state, { type, payload } )`

**Review by junaidshaikh-js — APPROVED** [review]
Good Work! 

Suggested on change in the comments, everything else is good. Can be merged.

File: src/context/user-products-context.jsx (hunk 4 above)
Comment (abhij1607) [reply src/context/user-products-context.jsx:9]: okay, will do it later during abstraction

**Pull request merged**